### PR TITLE
Fix double requests when search in order and product list

### PR DESCRIPTION
**Pull request by poulch**
I want to merge this change because it fixes two requests in order and product list views.

The main problem that we have was that when we want to search by query in those views we have to change the sort type to rank and descending. So when you enter the order or product view and try to search something, that changes query params which causes the component to rerender and request with those data that has been sent:
```
 "sort": {
    "direction": "ASC",
    "field": "NAME"
  }
  ```
  
  Then we used `useSortRedirect` hook that change sorting parameters to align with API. This causes a second request with this data:
  
  ```
   "sort": {
    "direction": "DESC",
    "field": "RANK"
  },
  ```

Steps to reproduce:
1. Go to product or order views
2. Open Graphql dev tools
3. Try to search for something
4. You should see the double requests in dev tools due to the reason mentioned above 

AC:
1. [ ] Only one request is sent when the user uses a search query
1. [ ] When the user  clear the search query should have previously selected the sorting direction 

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
5. [ ] All visible strings are translated with proper context including data-formatting
6. [ ] Attributes `[data-test-id]` are added for new elements
7. [ ] Changes are mentioned in the changelog
8. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
9. [ ] productType
10. [ ] shipping
11. [ ] customer
12. [ ] permissions
13. [ ] menuNavigation
14. [ ] pages
15. [ ] sales
16. [ ] vouchers
17. [ ] homePage
18. [ ] login
19. [ ] orders
20. [ ] products
21. [ ] app
22. [ ] plugins
23. [ ] translations
24. [ ] navigation
25. [ ] variants
26. [ ] payments

CONTAINERS=1

### CODE REVIEW

File: src/hooks/useFilterHandlers.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    const sortWithoutQuery =
+      params.sort === "rank" ? defaultSortField : params.sort;
+    const sortWithQuery = "rank" as SortField;
+
+    const getAscParam = () => {
+      if (hasQuery) {
+        return false;
+      }
+
+      if (!hasQuery && prevAsc !== null) {
```
Comment (andrzejewsky) [inline src/hooks/useFilterHandlers.ts:110]: **suggestion**: if you reach this line that means hasQuery is already falsy, so you don't need !hasQuery

**Review by andrzejewsky — APPROVED** [review]

File: src/hooks/useFilterHandlers.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  Sort<any> &
+  Pagination & { presestesChanged?: string };
+type CreateUrl = (params: RequiredParams) => string;
+type CreateFilterHandlers<TFilterKeys extends string> = [
+  (filter: IFilter<TFilterKeys>) => void,
+  () => void,
+  (query: string) => void,
+];
+
+export const useFilterHandlers = <
```
Comment (krzysztofzuraw) [inline src/hooks/useFilterHandlers.ts:22]: **suggestion**: can you add unit tests for this hook?

**Review by krzysztofzuraw — APPROVED** [review]

**Review by andrzejewsky — DISMISSED** [review]

**Comment by michalina-graczyk** [comment]
@poulch please provide AC for this ticket, current title doesn't describe what should be tested here.

**Review by michalina-graczyk — APPROVED** [review]
QA approved

**Review by krzysztofzuraw — APPROVED** [review]

**Pull request merged**